### PR TITLE
Optimize the code: fix CallbackServiceCodec.java exportOrunexportCallbackService method issue.

### PR DESCRIPTION
**Pull request by zhaoyuguang**
## What is the purpose of the change

fix CallbackServiceCodec.java exportOrunexportCallbackService method issue.
1. method exportOrunexportCallbackService throw NullPointerException when parameter url is null;
2. private field need camel writing.

## Brief changelog

image
method exportOrunexportCallbackService code: 
String group = url.getParameter(Constants.GROUP_KEY);

## Verifying this change

image

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java [hunk 1]
```diff
@@ -77,49 +77,49 @@ private static byte isCallBack(URL url, String methodName, int argIndex) {
...
                 if (!isInstancesOverLimit(channel, url, clazz.getName(), instid, false)) {
-                    Invoker<?> invoker = proxyFactory.getInvoker(inst, clazz, exporturl);
+                    Invoker<?> invoker = proxyFactory.getInvoker(inst, clazz, exportUrl);
                     // should destroy resource?
                     Exporter<?> exporter = protocol.export(invoker);
                     // this is used for tracing if instid has published service or not.
                     channel.setAttribute(cacheKey, exporter);
-                    logger.info("export a callback service :" + exporturl + ", on " + channel + ", url is: " + url);
-                    increaseInstanceCount(channel, countkey);
+                    logger.info("export a callback service :" + exportUrl + ", on " + channel + ", url is: " + url);
```
Comment (lixiaojiee) [inline dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java]: would you please change 'exporter' of line 113 to 'Exporter' by the way?

Comment (zhaoyuguang) [reply dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java]: why not

Comment (zhaoyuguang) [reply dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java]: i has fix it.

File: dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java [hunk 2]
```diff
@@ -77,49 +77,49 @@ private static byte isCallBack(URL url, String methodName, int argIndex) {
...
                 if (!isInstancesOverLimit(channel, url, clazz.getName(), instid, false)) {
-                    Invoker<?> invoker = proxyFactory.getInvoker(inst, clazz, exporturl);
+                    Invoker<?> invoker = proxyFactory.getInvoker(inst, clazz, exportUrl);
                     // should destroy resource?
                     Exporter<?> exporter = protocol.export(invoker);
                     // this is used for tracing if instid has published service or not.
                     channel.setAttribute(cacheKey, exporter);
-                    logger.info("export a callback service :" + exporturl + ", on " + channel + ", url is: " + url);
-                    increaseInstanceCount(channel, countkey);
+                    logger.info("Exporter a callback service :" + exportUrl + ", on " + channel + ", url is: " + url);
```
Comment (ralf0131) [inline dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java]: Exporter is a noun, we should use a verb here, like `Export`

Comment (lixiaojiee) [reply dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java]: Haha,my fault

**Comment by zhaoyuguang** [comment]
Rogar~ I will better it.

File: dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java [hunk 3]
```diff
@@ -77,49 +77,49 @@ private static byte isCallBack(URL url, String methodName, int argIndex) {
...
         }
         // add method, for verifying against method, automatic fallback (see dubbo protocol)
         params.put(Constants.METHODS_KEY, StringUtils.join(Wrapper.getWrapper(clazz).getDeclaredMethodNames(), ","));
 
-        Map<String, String> tmpmap = new HashMap<String, String>(url.getParameters());
-        tmpmap.putAll(params);
-        tmpmap.remove(Constants.VERSION_KEY);// doesn't need to distinguish version for callback
-        tmpmap.put(Constants.INTERFACE_KEY, clazz.getName());
-        URL exporturl = new URL(DubboProtocol.NAME, channel.getLocalAddress().getAddress().getHostAddress(), channel.getLocalAddress().getPort(), clazz.getName() + "." + instid, tmpmap);
+        Map<String, String> tmpMap = new HashMap<String, String>(url.getParameters());
```
Comment (satansk) [inline dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java]: `new HashMap<>(...)` seems more concise to me.

Comment (zhaoyuguang) [reply dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java]: Yes, the generics that can be inferred automatically are also deleted.

File: dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java [hunk 4]
```diff
@@ -77,49 +77,49 @@ private static byte isCallBack(URL url, String methodName, int argIndex) {
...
         int instid = System.identityHashCode(inst);
 
-        Map<String, String> params = new HashMap<String, String>(3);
+        Map<String, String> params = new HashMap<>(3);
         // no need to new client again
         params.put(Constants.IS_SERVER_KEY, Boolean.FALSE.toString());
         // mark it's a callback, for troubleshooting
         params.put(Constants.IS_CALLBACK_SERVICE, Boolean.TRUE.toString());
-        String group = url.getParameter(Constants.GROUP_KEY);
+        String group = (url == null ? null : url.getParameter(Constants.GROUP_KEY));
```
Comment (ralf0131) [inline dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java]: I check the code, it is possible for url to be null. So nice job here!

**Pull request merged**